qoqo-iqm/src/backend.rs

```diff
@@ -84,26 +92,27 @@ impl BackendWrapper {
     #[pyo3(text_signature = "(device, access_token)")]
     #[new]
     pub fn new(device: &PyAny, access_token: Option<String>) -> PyResult<Self> {
-        let device = match DenebDeviceWrapper::from_pyany(device.into()) {
-            Ok(device) => IqmDevice::from(device),
-            Err(_) => match ResonatorFreeDeviceWrapper::from_pyany(device.into()) {
-                Ok(device) => IqmDevice::from(device),
-                Err(err) => {
-                    return Err(PyRuntimeError::new_err(format!(
-                    "Could not convert device to neither DenebDevice nor ResonatorFreeDevice: {:?}",
-                    err
-                )))
-                }
-            },
-        };
+        let iqm_device: IqmDevice;
+        if let Ok(dev) = DenebDeviceWrapper::from_pyany(device.into()) {
+            iqm_device = IqmDevice::from(dev);
+        } else if let Ok(dev) = GarnetDeviceWrapper::from_pyany(device.into()) {
+            iqm_device = IqmDevice::from(dev);
+        } else if let Ok(dev) = ResonatorFreeDeviceWrapper::from_pyany(device.into()) {
+            iqm_device = IqmDevice::from(dev);
```

Comment:
While this code looks fine, I would rename the `dev` variable to something slightly more verbose, maybe `device` or `res`?